### PR TITLE
Update pure.css to v2.1.0

### LINKED ISSUE #1873: vendored normalize.css does strange things to html details/summary tags
Consider self hosted documentation, it is clear that usage examples are expandable
https://pacak.github.io/bpaf/bpaf/fn.short.html

The same function as hosted on docs.rs - it is NOT clear that they are expandable.
https://docs.rs/bpaf/0.6.1/bpaf/fn.short.html

tags in question

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

css in question is hosted here: https://docs.rs/-/static/vendored.css?0-6-0-b194f43-2022-09-25
```css
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
main,
menu,
nav,
section,
summary /* <- this part */ {
 display:block
}
```

### PR DESCRIPTION
**Pull request by Nemo157**
This updates normalize.css from v3.0.2 -> v8.0.1 ([changelog](https://github.com/necolas/normalize.css/blob/master/CHANGELOG.md)) and pure.css from v0.6.0 -> v2.1.0 ([changelog](https://github.com/pure-css/pure/blob/master/HISTORY.md)). In general the normalize.css breaking changes are to remove their opinionated styles and go towards a more generic "just fix the browser inconsistencies" along with dropping support for old browsers (still completely covering the [rust-lang supported browsers](https://rust-lang.github.io/rfcs/1985-tiered-browser-support.html)). The pure.css breaking changes just appear to be the updated normalize.css, changes to the build systems which don't affect us, and removing one deprecated class we don't use.

I've manually clicked through every page and haven't noticed any styling issues, in both desktop and emulated phone display.

fixes #1873

### CODE REVIEW

**Comment by syphar** [comment]
cc @GuillaumeGomez could you help with the review here?

File: vendor/pure-css/css/grids-responsive-min.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*!
-Pure v0.6.0
-Copyright 2014 Yahoo! Inc. All rights reserved.
+Pure v2.1.0
+Copyright 2013 Yahoo!
```
Comment (GuillaumeGomez) [inline vendor/pure-css/css/grids-responsive-min.css:3]: Funny that the copyright is going backward.

**Comment by GuillaumeGomez** [comment]
When I update the style with these changes, the UI changes too:

Before:
image

After:
Screenshot from 2022-10-16 19-51-59

This isn't expected right?

**Comment by Nemo157** [comment]
Nope, it should look just the same, which it does for me on my instance https://docs.rs.dev.nemo157.com/ | https://ipvold-docs-rs-dev.nemo157.com/ (I'll leave this PR running on it for at least the next day).

**Comment by GuillaumeGomez** [comment]
Strange. Well, if there is no UI changes, all good for me!

**Review by GuillaumeGomez — APPROVED** [review]

File: vendor/pure-css/css/grids-responsive-min.css (hunk 1 above)
Comment (Nemo157) [reply vendor/pure-css/css/grids-responsive-min.css:3]: Huh, I hadn't noticed that. I also updated [the `LICENSE` file](https://github.com/rust-lang/docs.rs/blob/master/vendor/pure-css/LICENSE) but that hadn't changed between versions, and that was already set to 2013.

**Pull request merged**